File: AI_Chat_Sever/AI_Chat_Sever/DialoguePolicy.cpp
#include "DialoguePolicy.h"

bool DialoguePolicy::AllowLLMCall(
    const CharacterState&,
    const DialogueSituation& situation)
{
    // 이벤트 시작 시만 허용
    return situation.event == "talk_start";
}

std::string DialoguePolicy::MakeCacheKey(
    const CharacterState& state,
    const DialogueSituation& situation)
{
    return state.characterId + "_" +
        std::to_string(state.relationship.affection / 20) + "_" +
        situation.event + "_" +
        situation.location;
}

DialogueResult DialoguePolicy::GetFallbackDialogue(
    const CharacterState&)
{
    return {
        "...",
        "neutral",
        "flat"
    };
}

File: AI_chat_Sever_PY/ai_service/policy/dialogue_policy.py
def allow_llm_call(req) -> bool:
    # 대화 시작 이벤트만 허용
    return req.situation.event == "talk_start"

def make_cache_key(req) -> str:
    stage = req.state.affection // 20
    return f"{req.state.character_id}:{stage}:{req.situation.event}:{req.situation.location}"

def fallback_dialogue(req):
    return {
        "dialogue": "...",
        "emotion": "neutral",
        "tone": "flat"
    }